### PR TITLE
Jetpack E2E: add block tests.

### PR DESCRIPTION
**Pull request by worldomonation**
Related to: https://github.com/Automattic/wp-calypso/issues/80730.

## Proposed Changes

This PR adds multiple block tests categorized under Important tasks.

Key flows/blocks:
- GIF block
- Map block
- Image compare block
- Story block

## Testing Instructions

Ensure the following build configurations are passing:
  - [x] Calypso E2E (desktop)
  - [x] Jetpack E2E Simple (mobile)
  - [x] Jetpack E2E Simple (desktop)
  - [ ] Jetpack E2E AT (desktop)

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: test/e2e/specs/blocks/blocks__story.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+/**
+ * Isolated block test for the Story block due to accessiblity issues,
+ * making it unable to be run using the BlockFlow pattern.
+ *
+ * @see https://github.com/Automattic/jetpack/issues/32976
+ *
+ * Keywords: Jetpack, Media Block, Story
+ */
+describe( DataHelper.createSuiteTitle( 'Blocks: Jetpack Story' ), function () {
```
Comment (worldomonation) [inline test/e2e/specs/blocks/blocks__story.ts:30]: As described below, the Story block must be in its own standalone spec due to issues with the block that affects the BlockFlow pattern.

File: test/e2e/specs/blocks/blocks__jetpack-media.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	TiledGalleryBlockFlow,
+	ImageCompareFlow,
+} from '@automattic/calypso-e2e';
+import { TEST_IMAGE_PATH, ALT_TEST_IMAGE_PATH } from '../constants';
+import { createBlockTests } from './shared/block-smoke-testing';
+
+const blockFlows: BlockFlow[] = [
+	new SlideshowBlockFlow( { imagePaths: [ TEST_IMAGE_PATH, ALT_TEST_IMAGE_PATH ] } ),
+	new TiledGalleryBlockFlow( { imagePaths: [ TEST_IMAGE_PATH, ALT_TEST_IMAGE_PATH ] } ),
+	new ImageCompareFlow( { image1Path: TEST_IMAGE_PATH, image2Path: ALT_TEST_IMAGE_PATH } ),
```
Comment (worldomonation) [inline test/e2e/specs/blocks/blocks__jetpack-media.ts:17]: The `TiledGalleryBlockFlow` and `SlideshowBlockFlow` have been split out from the `test/e2e/specs/blocks/blocks__jetpack-other.ts` file.

File: packages/calypso-e2e/src/lib/blocks/story-block.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import { Page } from 'playwright';
+import { TestFile } from '../../types';
+import { EditorComponent } from '../components';
+
+/**
+ * Class representing the flow of using a Story block in the editor.
+ */
+export class StoryBlock {
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/blocks/story-block.ts:8]: Story block is implemented using the older Block POM pattern, due to the issue (mentioned throughout this PR).

File: test/e2e/specs/blocks/blocks__jetpack-other.ts [hunk 4]
```diff
@@ -4,17 +4,22 @@
...
+	envVariables,
 } from '@automattic/calypso-e2e';
-import { TEST_IMAGE_PATH, ALT_TEST_IMAGE_PATH } from '../constants';
 import { createBlockTests } from './shared/block-smoke-testing';
 
 const blockFlows: BlockFlow[] = [
-	new SlideshowBlockFlow( { imagePaths: [ TEST_IMAGE_PATH, ALT_TEST_IMAGE_PATH ] } ),
-	new TiledGalleryBlockFlow( { imagePaths: [ TEST_IMAGE_PATH, ALT_TEST_IMAGE_PATH ] } ),
 	new StarRatingBlock( { rating: 3.5 } ),
+	new GifFlow( { query: 'https://giphy.com/embed/MDJ9IbxxvDUQM' } ),
```
Comment (karenroldan) [inline test/e2e/specs/blocks/blocks__jetpack-other.ts:16]: **praise:** good choice of GIF 😄

File: test/e2e/specs/blocks/blocks__story.ts (hunk 1 above)
Comment (karenroldan) [inline test/e2e/specs/blocks/blocks__story.ts:30]: **nitpick:** We could remove the usage of `DataHelper.createSuiteTitle`

**Review by karenroldan — APPROVED** [review]
LGTM! 🚀 tests ran nicely locally. 👍🏻

File: test/e2e/specs/blocks/blocks__story.ts (hunk 1 above)
Comment (worldomonation) [reply test/e2e/specs/blocks/blocks__story.ts:30]: We now use the `createSuiteTitle` for AT tests to generate names that contain the AT environment, so turns out this is no longer deprecated, at least for Jetpack AT tests.

**Pull request merged**